android/modules/ui/src/java/ti/modules/titanium/ui/widget/tabgroup/TiUIAbstractTabGroup.java

```diff
@@ -60,12 +326,140 @@ public void processProperties(KrollDict d)
 			Object activityObject = d.get(TiC.PROPERTY_ACTIVITY);
 			ActivityProxy activityProxy = getProxy().getActivityProxy();
 			if (activityObject instanceof HashMap<?, ?> && activityProxy != null) {
-				@SuppressWarnings("unchecked")
 				KrollDict options = new KrollDict((HashMap<String, Object>) activityObject);
 				activityProxy.handleCreationDict(options);
 			}
 		}
-
+		if (d.containsKey(TiC.PROPERTY_SWIPEABLE)) {
+			this.swipeable = d.getBoolean(TiC.PROPERTY_SWIPEABLE);
+		}
+		if (d.containsKey(TiC.PROPERTY_SMOOTH_SCROLL_ON_TAB_CLICK)) {
+			this.smoothScrollOnTabClick = d.getBoolean(TiC.PROPERTY_SMOOTH_SCROLL_ON_TAB_CLICK);
+		}
 		super.processProperties(d);
 	}
+
+	@Override
+	public void propertyChanged(String key, Object oldValue, Object newValue, KrollProxy proxy)
+	{
+		if (key.equals(TiC.PROPERTY_SWIPEABLE)) {
+			this.swipeable = TiConvert.toBoolean(newValue);
+		} else if (key.equals(TiC.PROPERTY_SMOOTH_SCROLL_ON_TAB_CLICK)) {
+			this.smoothScrollOnTabClick = TiConvert.toBoolean(newValue);
+		} else {
+			super.propertyChanged(key, oldValue, newValue, proxy);
+		}
+	}
+
+	/**
+	 * Returns the currently selected TabProxy
+	 * @return the TabProxy instance.
+	 */
+	public TabProxy getSelectedTab()
+	{
+		return ((TabGroupProxy) getProxy()).getTabList().get(this.tabGroupViewPager.getCurrentItem());
+	}
+
+	/**
+	 * Implemenation of the FragmentPagerAdapter
+	 */
+	private class TabGroupFragmentPagerAdapter extends FragmentPagerAdapter
+	{
+
+		public TabGroupFragmentPagerAdapter(FragmentManager fm)
+		{
+			super(fm);
+		}
+
+		// getItem only gets called by the FragmentPagerAdapter when the fragment is not found
+		// in the FragmentManager. We construct it and associate it to the tab view.
+		@Override
+		public Fragment getItem(int i)
+		{
+			TabFragment tabFragment = new TabFragment();
+			tabFragment.setTab(tabs.get(i));
+			return tabFragment;
+		}
+
+		// Android docs say we must override the default implementation if item position can change
+		// The FragmentPagerAdapter uses the ID to construct the fragment tag (android:switcher:containerID:ID)
+		// and then checks the FragmentManager for presence of said tag. The default Android implementation
+		// returns the fragment position in the ViewPager, which is of course wrong when a tab is removed
+		// and fragments in higher positions move. Thus we maintain the position and IDs in an ArrayList ourselves.
+		@Override
+		public long getItemId(int position)
+		{
+			long id = tabs.get(position).hashCode();
+			return id;
```

Comment:
The returned hash code is not guaranteed to be unique.
How did this work before?

Reply:
It had an AtomicLong that generated IDs for each added tab. I will push an update with a simplified flow like it worked before.